core/commands/remotepin.go

```diff
@@ -0,0 +1,480 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	cid "github.com/ipfs/go-cid"
+	cmds "github.com/ipfs/go-ipfs-cmds"
+	config "github.com/ipfs/go-ipfs-config"
+	"github.com/ipfs/go-ipfs/core/commands/cmdenv"
+	fsrepo "github.com/ipfs/go-ipfs/repo/fsrepo"
+	pinclient "github.com/ipfs/go-pinning-service-http-client"
+	path "github.com/ipfs/interface-go-ipfs-core/path"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+	"github.com/multiformats/go-multiaddr"
+)
+
+var remotePinURL = os.Getenv("IPFS_REMOTE_PIN_SERVICE")
+var remotePinKey = os.Getenv("IPFS_REMOTE_PIN_KEY")
+
+var remotePinCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Pin (and unpin) objects to remote pinning service.",
+	},
+
+	Subcommands: map[string]*cmds.Command{
+		"add":     addRemotePinCmd,
+		"ls":      listRemotePinCmd,
+		"rm":      rmRemotePinCmd,
+		"service": remotePinServiceCmd,
+	},
+}
+
+var remotePinServiceCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Configure remote pinning services.",
+	},
+
+	Subcommands: map[string]*cmds.Command{
+		"add":    addRemotePinServiceCmd,
+		"rename": renameRemotePinServiceCmd,
+		"update": updateRemotePinServiceCmd,
+		"rm":     rmRemotePinServiceCmd,
+	},
+}
+
+const pinNameOptionName = "name"
+const pinCIDsOptionName = "cid"
+const pinServiceNameOptionName = "service"
+
+type AddRemotePinOutput struct {
```

Comment:
@rafaelramalho19 noted that this struct  is used not only in `add` but also `ls`, which makes it confusing on the first glance. 

Perhaps should be renamed 
```suggestion
type RemotePin struct {
```

Thoughts?

Reply:
I renamed it to RemotePinOutput ... seems more descriptive than RemotePin, and follows some implicit naming convention used in other commands.